### PR TITLE
[eventd] Add incremental polling when waiting for capture service to start

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Why I did it

Addresses https://github.com/sonic-net/sonic-buildimage/issues/17350

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Instead of a 1 second delay, we poll to check that the thread is available and after each poll increment the delay. There were situations where if there was less memory available, fixed polling would not be effective for starting zmq capture service. Add an incremental delay such that eventd can wait longer to start up capture service if system is too busy or overloaded, but still keep a max duration/retry limit so that we do not wait forever.

#### How to verify it

UT

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by dgsudharsan** [comment]
@zbud-msft  Can you please check if we can cherrypick this fix to 202305? If not can you pelase create a separate PR?

File: src/sonic-eventd/src/eventd.cpp [hunk 1]
```diff
@@ -547,8 +548,9 @@ capture_service::set_control(capture_control_t ctrl, event_serialized_lst_t *lst
         case INIT_CAPTURE:
             m_thr = thread(&capture_service::do_capture, this);
             for(int i=0; !m_cap_run && (i < CAPTURE_SERVICE_POLLING_RETRIES); ++i) {
-                /* Wait max a second for thread to init */
-                this_thread::sleep_for(chrono::milliseconds(CAPTURE_SERVICE_POLLING_DURATION));
+                /* Poll to see if thread has been init, if so exit early. Add delay on every attempt */
```
Comment (qiluo-msft) [inline src/sonic-eventd/src/eventd.cpp:551]: >Poll [](http://example.com/codeflow?start=19&length=4)

It reduces the possibility of the issue happening, and it is not a completed fix. Then we need to handle the issue as a warning, not as an error.

**Comment by StormLiangMS** [comment]
hi @qiluo-msft  @zbud-msft  could you update the ADO? Is this a must to have for 202305?

**Comment by dgsudharsan** [comment]
@zbud-msft  This PR has conflict with 202305. Can you please check and if required create a separate PR for 202305?

**Comment by zbud-msft** [comment]
> @zbud-msft This PR has conflict with 202305. Can you please check and if required create a separate PR for 202305?

@dgsudharsan Currently depending on https://github.com/sonic-net/sonic-buildimage/pull/18249

File: src/sonic-eventd/src/eventd.cpp (hunk 1 above)
Comment (zbud-msft) [reply src/sonic-eventd/src/eventd.cpp:551]: I added logic that eventd will not exit when capture service fails to initialize; warning logs are added, and main proxy/zmq service will continue to run.

Comment (qiluo-msft) [reply src/sonic-eventd/src/eventd.cpp:551]: closed.

**Review by qiluo-msft — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@yxieca  Can you please help to cherry-pick this fix to 202311?

**Pull request merged**